zebra/interface.c

```diff
@@ -1483,23 +1483,27 @@ static void interface_vrf_change(enum dplane_op_e op, ifindex_t ifindex,
 				"DPLANE_OP_INTF_UPDATE for VRF %s(%u) table %u",
 				name, ifindex, tableid);
 
-		if (!vrf_lookup_by_id((vrf_id_t)ifindex)) {
-			vrf_id_t exist_id;
-
-			exist_id = zebra_vrf_lookup_by_table(tableid, ns_id);
-			if (exist_id != VRF_DEFAULT) {
-				vrf = vrf_lookup_by_id(exist_id);
-
-				if (vrf)
-					flog_err(EC_ZEBRA_VRF_MISCONFIGURED,
-						 "VRF %s id %u table id overlaps existing vrf %s(%d), misconfiguration exiting",
-						 name, ifindex, vrf->name,
-						 vrf->vrf_id);
-				else
-					flog_err(EC_ZEBRA_VRF_NOT_FOUND,
-						 "VRF %s id %u does not exist",
-						 name, ifindex);
+		/*
+		 * For a given tableid, if there already exists a vrf and it
+		 * is different from the current vrf to be operated, then there
+		 * is a misconfiguration and zebra will exit.
+		 */
+		vrf_id_t exist_id = zebra_vrf_lookup_by_table(tableid, ns_id);
+
+		if (exist_id != VRF_DEFAULT) {
+			vrf = vrf_lookup_by_id(exist_id);
+
+			if (!vrf_lookup_by_id((vrf_id_t)ifindex) && !vrf) {
+				flog_err(EC_ZEBRA_VRF_NOT_FOUND,
+					 "VRF %s id %u does not exist", name,
+					 ifindex);
+				exit(-1);
```

Comment:
Just for my understanding: why is this exit, and not return?

Reply:
To keep inline with the existing code. 
I think you introduced this in (https://github.com/FRRouting/frr/commit/4bd04364adb27e18052730a54ee9a85ae12386c2) , although not sure how easy it is to hit this condition.

Reply:
I was about `exit()`, not about NULL check.

Reply:
yes, i am talking about exit()

In the existing code, even if vrf was NULL, we exit()
```
			if (exist_id != VRF_DEFAULT) {
				vrf = vrf_lookup_by_id(exist_id);

				if (vrf)
					flog_err(EC_ZEBRA_VRF_MISCONFIGURED,
						 "VRF %s id %u table id overlaps existing vrf %s(%d), misconfiguration exiting",
						 name, ifindex, vrf->name,
						 vrf->vrf_id);
				else
					flog_err(EC_ZEBRA_VRF_NOT_FOUND,
						 "VRF %s id %u does not exist",
						 name, ifindex);

				exit(-1);
			}
```